app/services/mongodb/MongoDBFileService.scala

```diff
@@ -814,67 +816,74 @@ class MongoDBFileService @Inject() (
   def removeFile(id: UUID, host: String, apiKey: Option[String], user: Option[User]){
     get(id) match{
       case Some(file) => {
-        if(!file.isIntermediate){
-          val fileDatasets = datasets.findByFileIdDirectlyContain(file.id)
-          for(fileDataset <- fileDatasets){
-            datasets.removeFile(fileDataset.id, id)
-            if(!file.xmlMetadata.isEmpty){
-              datasets.index(fileDataset.id)
-            }
+        val currentResourceRef = ResourceRef(ResourceRef.file, file.id)
+        val hasPermission = Permission.checkPermission(user.get,Permission.DeleteFile, currentResourceRef)
```

Comment:
This call is redundant and has already been done in most cases for this function.